pallets/address-associations/src/lib.rs

```diff
@@ -0,0 +1,101 @@
+//! Pallet storing associations from main chain addresses to parter chain address.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+	use scale_info::prelude::fmt::Debug;
+	use sp_std::vec::Vec;
+
+	pub const PALLET_VERSION: u32 = 1;
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// Main chain address type
+		type MainChainAddress: Member + Parameter + MaxEncodedLen;
+
+		/// Partner chain address type
+		type PartnerChainAddress: Member + Parameter + MaxEncodedLen;
```

Comment:
I would make it a `Value` and make the pallet cardano-associations. I don't even know if cardano is important part here.

Reply:
I don't think Cardano is important here, but having a `key-pair-pallet` would be too general. I'd like to stay within the domain "associating some main chain identifier with some partner chain identifier", whatever chain the main chain and partner chain happen to be.